pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -160,51 +188,132 @@ const CryptoPolicyDialog = ({
+    const [fipsEnabled, setFipsEnabled] = useState(null);
     const [showShutdownModal, setShowShutDownModal] = useState(false);
+    const [showEnableFIPSModal, setShowEnableFIPSModal] = useState(false);
 
     useEffect(() => {
         if (!requiresReboot) {
             // For when we change the setting
             cockpit.file(applyNeedRebootStamp).watch(content => setRequiresReboot(content !== null));
         }
-    }, [requiresReboot]);
 
-    if (!requiresReboot) {
-        return null;
+        // Avoid cockpit.spawn every re-render from parent
+        if (currentCryptoPolicy === null) {
+            cockpit.spawn(["update-crypto-policies", "--show"], { err: "message" })
+                    .then(output => setCurrentCryptoPolicy(output.trim()))
+                    .catch(err => {
+                        console.debug("no crypto policies support: ", err.toString());
+                        setCurrentCryptoPolicy(undefined);
+                    });
+            // When users run `update-crypto-policies` manually.
+            cockpit.file("/etc/crypto-policies/state/current")
+                    .watch(content => setCurrentCryptoPolicy(content ? content.trim() : null));
+        }
+
+        if (fipsEnabled === null)
+            cockpit.file("/proc/sys/crypto/fips_enabled").read()
+                    .then(content => setFipsEnabled(content ? content.trim() === "1" : false));
+    }, [currentCryptoPolicy, fipsEnabled, requiresReboot]);
+
+    if (requiresReboot) {
+        return (
+            <li className="system-health-crypto-policies">
+                <Flex flexWrap={{ default: 'nowrap' }} spaceItems={{ default: 'spaceItemsSm' }} alignItems={{ default: 'alignItemsCenter' }}>
+                    <InProgressIcon size="sm" className="crypto-policies-health-card-icon" />
+                    <Button isInline variant="link" onClick={() => setShowShutDownModal(true)}>
+                        {_("Reboot to apply new crypto policy")}
+                    </Button>
+                </Flex>
+                {showShutdownModal &&
+                <ShutdownModal onClose={() => setShowShutDownModal(false)} />
+                }
+            </li>
+        );
     }
 
+    if (!requiresReboot && currentCryptoPolicy === "FIPS" && !fipsEnabled) {
+        return (
+            <li className="system-health-crypto-policies">
+                <Flex spacer={{ default: 'spaceItemsSm' }} flexWrap={{ default: 'nowrap' }}>
+                    <FlexItem><ExclamationTriangleIcon size="sm" className="crypto-policies-health-card-icon" /></FlexItem>
+                    <div>
+                        <div id="fips_mode_disabled_text">
+                            {_("FIPS crypto policy selected, but not active")}
+                        </div>
+                        <Button isInline variant="link" className="pf-u-font-size-sm" onClick={() => setShowEnableFIPSModal(true)}>
+                            {_("Enable FIPS mode and reboot")}
```

Comment:
This sounds a bit scary, like it would immediately do this. Can we take out the "and reboot" here? (Also see below)

Reply:
We do, but the text does not make this clear. We also don't use the "..." convention to indicate that some action opens a dialog first.